### DISCUSSION TITLE
Faulty `char != -1` check in `OracleDialectRules.QStringRule.doEvaluate`

### DISCUSSION THREAD

**Discussion by Marcono1234** [dbeaver/dbeaver#19898] [DBeaver Development] [answered]
### Description

`OracleDialectRules.QStringRule.doEvaluate` contains the following check:
https://github.com/dbeaver/dbeaver/blob/0642e8c736bb919041add49621b0cc4a819838bb/plugins/org.jkiss.dbeaver.ext.oracle/src/org/jkiss/dbeaver/ext/oracle/sql/OracleDialectRules.java#L55

The problem is that `quoteStartChar` is a `char` and the minimum value of `char` is 0, so the `!= -1` check will always be true.

Maybe the simplest solution is to change `quoteStartChar` to type `int` and remove the `char` cast.

I am not planning to submit a pull request though because I am not familiar with this project.

### DBeaver Version

commit 0642e8c736bb919041add49621b0cc4a819838bb

**Comment by E1izabeth** [comment]
Have you meet some bug? What issue are you trying to solve?

**Reply by E1izabeth** [reply]
Anyway, thank you for interest to our project and finding a flaw.

**Reply by Marcono1234** [reply]
No sorry I haven't investigated the exact consequences of this bug. I was investigating this general bug pattern when I stumbled upon it in your repository and thought it would at least be useful to let you know about it, because it has for sure some (negative) consequence.

**Comment by Marcono1234** [comment, marked as answer]
For future reference, it looks like this was fixed by https://github.com/dbeaver/dbeaver/pull/20151
